### PR TITLE
Fix a number of issues with updating celery, bugs in disk space check

### PR DESCRIPTION
**Pull request by dmick**
1) really fix the checks.py call for celery_has_workers() (not Inspect(), but app.control.inspect())
2) fix up disk_has_space checks: wrong config item name, popen() result not waited for, Py3 stdout/err are binary
3) force use of py-amqp rather than librabbitmq; the latter is apparently deprecated and is causing problems.
4) improve disk_has_space: check both repos_root and binary_root
5) (added late): fix binary checksum calculation for Py3.

### CODE REVIEW

**Comment by dmick** [comment]
@djgalloway fyi.  I think I want to discuss the last commit with @andrewschoen before changing this out of draft

**Comment by andrewschoen** [comment]
> @djgalloway fyi. I think I want to discuss the last commit with @andrewschoen before changing this out of draft

I took a look at https://github.com/ceph/chacra/pull/301/commits/ea1a489a28537f88d09e9b5ede552c2ddaca0835 

They only reason for using `librabbitmq` that I remember is because the docs recommended it. Reading through the ticket you mentioned in the commit and this confirms that we should not be using `librabbitmq` anymore: https://github.com/celery/celery/pull/6933/commits/38b0527b170408f1d86bd48890fa7e47a35f9baf

**Review by andrewschoen — APPROVED** [review]

**Comment by dmick** [comment]
Added another commit that had been on another pending branch to fix binary checksumming; this was preventing binary uploads.  It really should have been part of the python3 port, but the code wasn't exercised until I got a test instance up and running, and then I probably should have put it on a different branch, but it was on the wip-add-raw-repos branch because that's where I tested it.

Anyway, it's on this branch now and will resolve the current issue with 1.chacra.ceph.com.

File: chacra/asynch/__init__.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def bootstrap_pecan(signal, sender, **kw):
 
 app = Celery(
     'chacra.asynch',
-    broker='amqp://guest@localhost//',
+    broker='pyamqp://guest@localhost//',
```
Comment (djgalloway) [inline chacra/asynch/__init__.py:66]: wow...

**Pull request merged**